core/get_ratings.py

```diff
@@ -1,148 +1,141 @@
-    set_counts[winner] += total_sets
-    set_counts[looser] += total_sets
+    new_ratings.loc[players.winner].Rating = game_ratings_after.my
+    new_ratings.loc[players.looser].Rating = game_ratings_after.opponent
+    set_counts[players.winner] += score.winner + score.looser
+    set_counts[players.looser] += score.winner + score.looser
     return new_ratings
 
 
 def count_ratings(date):
-    game_data_filepath = os.path.join(repo_root_dir, 'data/games.csv')
+    game_data_filepath = os.path.join(REPO_ROOT_DIR, 'data/games.csv')
     games = pd.read_csv(game_data_filepath)
     sorted_games = games.sort_values(['Date', 'Game of the day'])
     proper_games = sorted_games[sorted_games['Date'] <= date]
+
     all_players = list(set(proper_games['Winner'].tolist() + proper_games['Looser'].tolist()))
     data = {'Rating': [constants.DEFAULT_RATING for i in range(len(all_players))]}
     start_ratings = pd.DataFrame(data, index=all_players)
     n_games = proper_games.shape[0]
     set_counts = {name: 0 for name in all_players}
     ratings_log = [(proper_games.loc[0]['Date'], start_ratings)]
+
     for i in range(n_games):
         new_ratings = update_ratings_with_game(ratings_log[-1][1], proper_games.loc[i], set_counts)
         new_ratings = new_ratings.sort_values(['Rating'], ascending=False)
         ratings_log.append((proper_games.loc[i]['Date'], new_ratings.copy(deep=True)))
     return ratings_log[-1][1], ratings_log, set_counts
 
 
+def get_previous_rating(name, ratings_log):
+    for i in range(len(ratings_log)):
+        cur_day, cur_ratings = ratings_log[i]
+        if cur_day == ratings_log[-1][0]:
+            if i != 0:
+                prev_day, prev_ratings = ratings_log[i-1]
+                return int(np.round(prev_ratings.loc[name]['Rating']))
+            else:
+                return int(np.round(cur_ratings.loc[name]['Rating']))
+
+
 def save_ratings_to_json(ratings, ratings_log, set_counts):
-    json_list = []
+    json_data = []
     for name in ratings.index:
         dict_to_save = {'name': name,
                         'rating': int(np.round(ratings.loc[name]['Rating'])),
-                        'sets': int(set_counts[name])}
-        for i in range(len(ratings_log)):
-            cur_date, cur_ratings = ratings_log[i]
-            if cur_date == ratings_log[-1][0]:
-                if i != 0:
-                    prev_day, prev_ratings = ratings_log[i-1]
-                    dict_to_save['prev_rating'] = int(np.round(prev_ratings.loc[name]['Rating']))
-                else:
-                    dict_to_save['prev_rating'] = int(np.round(cur_ratings.loc[name]['Rating']))
-                break
-        json_list.append(dict_to_save)
-    json_filename = os.path.join(repo_root_dir, 'leaderboard-ui/src/rating.json')
+                        'sets': int(set_counts[name]),
+                        'prev_rating': get_previous_rating(name, ratings_log)}
+        json_data.append(dict_to_save)
```

Comment:
could you replace it with `map`?

Reply:
Did not understand. Do you mean `[value for _ in _]` syntax?

Reply:
Wouldn't it be too complicated?

Reply:
```python
arr = []
for ...
	arr.push()
```

is equal to purest version:

```python
arr = list(map())
```

Reply:
So I have to implement dict extractor and use `list(map(my_func, data_to_iterate)`?

Reply:
```python
json_data = list(map(lambda name: {
	'name': name,
	'rating': int(np.round(ratings.loc[name]['Rating'])),
	'sets': int(set_counts[name]),
	'prev_rating': get_previous_rating(name, ratings_log)
}, ratings.index))
```